src/routes/user/dto/create-user.dto.ts

```diff
@@ -0,0 +1,33 @@
+import { IsEmail, IsNotEmpty, IsString, MinLength } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class CreateUserDto {
+  @ApiProperty({
+    description: 'User email address',
+    example: 'user@example.com',
+  })
+  @IsEmail()
+  @IsNotEmpty()
+  email: string;
+
+  @ApiProperty({ description: 'User username', example: 'xX_DarkGamer_Xx' })
+  @IsString()
+  @IsNotEmpty()
+  username: string;
+
+  @ApiProperty({ description: 'User first name', example: 'John' })
+  @IsString()
+  @IsNotEmpty()
+  firstName: string;
```

Comment:
Not sure if this is useful for our scenario. Maybe a proper `nickname` that e.g. allows for spaces in it?